src/io/local_storage.ts

```diff
@@ -0,0 +1,227 @@
+ * See the doc string to `browserLocalStorage` for more details.
+ */
+export class BrowserLocalStorage implements IOHandler {
+  protected readonly LS: Storage;
+  protected readonly modelPath: string;
+  protected readonly paths: {[key: string]: string};
+
+  constructor(modelPath: string) {
+    if (!(window && window.localStorage)) {
+      throw new Error(
+          'The current environment does not support local storage.');
+    }
+    this.LS = window.localStorage;
+
+    if (modelPath == null) {
+      throw new Error(
+          'For local storage, modelPath must not be null or undefined.');
+    }
+    if (!modelPath) {
+      throw new Error('For local storage, modelPath must not be empty.');
+    }
+    this.modelPath = modelPath;
+    const modelRoot = [PATH_PREFIX, this.modelPath].join(PATH_SEPARATOR);
+    this.paths = {
+      info: [modelRoot, INFO_SUFFIX].join(PATH_SEPARATOR),
+      topology: [modelRoot, MODEL_TOPOLOGY_SUFFIX].join(PATH_SEPARATOR),
+      weightSpecs: [modelRoot, WEIGHT_SPECS_SUFFIX].join(PATH_SEPARATOR),
+      weightData: [modelRoot, WEIGHT_DATA_SUFFIX].join(PATH_SEPARATOR),
+    };
+  }
+
+  /**
+   * Save model artifacts to browser local storage.
+   *
+   * See the documentation to `browserLocalStorage` for details on the saved
+   * artifacts.
+   *
+   * @param modelArtifacts The model artifacts to be stored.
+   * @returns An instance of SaveResult.
+   */
+  async save(modelArtifacts: ModelArtifacts): Promise<SaveResult> {
+    if (!(window && window.localStorage)) {
+      throw new Error(
+          'The current environment does not support local storage.');
+    }
+
+    if (modelArtifacts.modelTopology instanceof ArrayBuffer) {
+      throw new Error(
+          'BrowserLocalStorage.save() does not support saving model topology ' +
+          'in binary formats yet.');
+    } else {
+      const topology = JSON.stringify(modelArtifacts.modelTopology);
+      const weightSpecs = JSON.stringify(modelArtifacts.weightSpecs);
+
+      const modelArtifactsInfo: ModelArtifactsInfo = {
+        dateSaved: new Date(),
+        modelTopologyType: 'KerasJSON',
+        modelTopologyBytes: stringByteLength(topology),
+        weightSpecsBytes: stringByteLength(weightSpecs),
+        weightDataBytes: modelArtifacts.weightData.byteLength,
+      };
+
+      try {
+        this.LS.setItem(this.paths.info, JSON.stringify(modelArtifactsInfo));
+        this.LS.setItem(this.paths.topology, topology);
+        this.LS.setItem(this.paths.weightSpecs, weightSpecs);
+        this.LS.setItem(
+            this.paths.weightData,
+            arrayBufferToBase64String(modelArtifacts.weightData));
+
+        return {modelArtifactsInfo};
+      } catch (err) {
+        // If saving failed, clean up all items saved so far.
+        for (const key in this.paths) {
+          this.LS.removeItem(this.paths[key]);
+        }
+
+        throw new Error(
+            `Failed to save model '${this.modelPath}' to local storage: ` +
```

Comment:
Does err contain any information that we can use to say why it failed?  Do we need to do size checks before we attempt to write?  Are there possible permissions issues we can pre-check?